### ISSUE TITLE
numRuns is ignored with fc.array()

### ISSUE THREAD

**Issue by brandon-leapyear** [dubzzz/fast-check#528]
## 🐛 Bug Report

It seems like when using `fc.array` and `numRuns`, `numRuns` is ignored and the test is run multiple times when the test passes.

## To Reproduce

```js
fc.assert(
  fc.property(fc.array(fc.string()), (v) => {
    console.log(v)
    return v === []
  }),
  { numRuns: 1 }
)
```

(can use `seed: 221016768, path: "0:0"` to see a repro of this bug, but running this enough times should also surface the bug)

output (with the above seed):
```
    [
      'JC7yf', 'S&q',
      '!9!g',  'o,wX[}',
      's',     'xd*O.qy',
      'L6K'
    ]
    []
```

## Expected behavior

It should only run once

## Your environment

| Packages / Softwares  | Version(s) |
| --------------------- | ---------- |
| fast-check            |   1.21.0         |
| node                  |    12.13.0        |
| TypeScript*           |   3.7.4         |

*Only for TypeScript's users

**Comment by brandon-leapyear** [comment]
It seems like the effect is compounded if i do things like
```js
fc.array(...).chain((arr) =>
  fc.tuple(
    fc.constant(arr),
    fc.subarray(arr)
  )
)
```

**Comment by brandon-leapyear** [comment]
Oh wait, it might be the shrinking that's causing it...?

**Comment by dubzzz** [comment]
Behaviour is expected even in case you provide both a seed, a numRuns and a path. You might want to use `endOnFailure: true` to achieve your needs as described in https://github.com/dubzzz/fast-check/blob/master/documentation/1-Guides/Tips.md#replay-after-failure

The motivation behind the current behaviour:
- `numRuns: 1` tells the framework to generate a single value
- `path: "0:0"` tells the framework to offset the generated as follow: take the first value (first 0), shrink it (:), start at the first shrunk value (0). Then starting from this value in the shrinking path, fast-check will execute the property. In your case 0:0 is the first logged value (it fails), so fast-check shrinks again towards [] at path 0:0:0 which succeeds. That's the end because 0:0 had only one shrunk value. Otherwise fast-check would have tried 0:0:1 and so on..

**Comment by brandon-leapyear** [comment]
Yes, you're right. But let's say that I'm debugging a failure and then fix the failure. I want to run this seed/path again, so i do:

```js
fc.assert(
  fc.property(fc.array(fc.string()), (v) => {
    console.log(v)
    return true
  }),
  { numRuns: 1, seed: -1799161243, path: "59:12:8", endOnFailure: true }
)
```
I expect it to only run once (because of `numRuns: 1`), but I get many `console.log`s. If this is expected behavior, I find the name "numRuns" to be misleading. Would there be a way to change the name to something like `numGen` and then add `maxRuns` to be a maximum limit on the number of fast check executions?

My use case is that one of our fast check tests takes a long time to run once (~500ms), so we limit it to a few `numRuns` so that it doesn't exceed Jest's 4500ms timeout (we don't care about exhaustively checking, just wanting to avoid testing hardcoded values). But when debugging failures, fast check will do more than the few runs we configure with `numRuns`, and it'll timeout with jest, returning a false negative.

**Comment by dubzzz** [comment]
@brandon-leapyear Actually there is a setting - `interruptAfterTimeLimit` - to achieve what you are looking for. Unfortunately, the setting was missing from the Tips section of the documentation of fast-check so I just added it: https://github.com/dubzzz/fast-check/pull/529

- `interruptAfterTimeLimit` has been designed for this specific need: stop a test execution when it takes too long. 
- In the contrary, `numRuns` is a way to customize the number of values that should be used against your code, it can be seen as: let's see if my code breaks if fast-check sends me {numRuns} random values, and if it does (my code will never 🤞) let's report me the smallest possible failure. Concerning the possibility to change this `numRuns` into `numGen` it is not currently part of my plan as it will break many users of the lib.

Basically my recommendation would be to setup your test runner - by following https://github.com/dubzzz/fast-check/blob/master/documentation/1-Guides/Tips.md#setup-global-settings - with this configuration:
```js
fc.configureGlobal({
  interruptAfterTimeLimit: 4000, // Default timeout in Jest 5000ms
  markInterruptAsFailure: true,  // When set to true, timeout during initial cases (1) will be marked as an error
                                 // When set to false, timeout during initial cases (1) will not be considered as a failure
});
// And override numRuns locally for the tests taht take too long
```

**Comment by brandon-leapyear** [comment]
Thank you!

**Comment by dubzzz** [comment]
Please let me know if it does not solve your problem ;)
